### PR TITLE
React Hook Form: Update to v 7.49.2

### PR DESCRIPTION
**Pull request by Clarity-89**
**What is this feature?**

Update the react-hook-form package to the latest version.

**Why do we need this feature?**

The version of RHF we have is a few years old. Updating to the new version has a few benefits:
- We get to use the new methods (like [resetField](https://react-hook-form.com/docs/useform/resetfield))
- In the latest versions of RHF the types were significantly improved, providing better type safety, which allowed to catch a few bugs already. 

**Special notes for your reviewer:**

I see that RHF is included in the renovate ignore list to avoid breaking the Form component API. I think we'd just deprecate the Form component from grafana/ui, which would remove the extra non ui-related dependency we need to maintain.

### CODE REVIEW

File: public/app/features/alerting/unified/utils/amroutes.ts
Comment (gillesdemey) [inline public/app/features/alerting/unified/utils/amroutes.ts:240]: This probably needs to change; we shouldn't pass validation for the repeat interval if no group interval was specified since that's an optional field.

I suggest defaulting the `groupInterval = ''` and removing the additional check for `groupInterval`

**Review by gillesdemey — CHANGES_REQUESTED** [review]
one change required, everything else LGTM from the Alerting side :)

File: public/app/features/correlations/types.ts
Comment (gelicia) [inline public/app/features/correlations/types.ts:33]: This isn't a data link. The type is basically "whatever would go into a query editor" which is anything extended off of `DataQuery`, so we need this to be generic.

File: public/app/features/alerting/unified/utils/amroutes.ts
Comment (gillesdemey) [reply public/app/features/alerting/unified/utils/amroutes.ts]: Resolved in https://github.com/grafana/grafana/pull/79493/commits/f24bf2fd4ffb221ae1c46c1c3d604c153cebe86b

**Review by gillesdemey — APPROVED** [review]
Addressed the Alerting-related concern in https://github.com/grafana/grafana/pull/79493/commits/f24bf2fd4ffb221ae1c46c1c3d604c153cebe86b – LGTM from our side

File: public/app/features/alerting/unified/components/mute-timings/MuteTimingTimeRange.tsx
Comment (juanicabanas) [inline public/app/features/alerting/unified/components/mute-timings/MuteTimingTimeRange.tsx:31]: I don't think this is necessary

**Review by juanicabanas — APPROVED** [review]
LGTM. Just a nit

**Review by ashharrison90 — APPROVED** [review]
from a code standpoint this lgtm 👍 

haven't tested behaviour in alerting/explore etc - i'll leave that to the relevant affected teams 

think it would be useful to confirm that this won't be a breaking change for e.g. a plugin using our form components

**Review by Elfo404 — APPROVED** [review]
Explore/Correlations looks good! 👍

File: public/app/features/alerting/unified/components/mute-timings/MuteTimingTimeRange.tsx
Comment (Clarity-89) [reply public/app/features/alerting/unified/components/mute-timings/MuteTimingTimeRange.tsx:31]: I get a type error without this change.

**Pull request merged**
